### DISCUSSION TITLE
Problem installing aws-sdk-php

### DISCUSSION THREAD

**Discussion by alonr619** [pimcore/pimcore#8339] [Q&A / Support] [answered]
Hi,

I am trying to connect Pimcore to AWS S3 using [this](https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Amazon_AWS_Setup/Amazon_AWS_S3_Setup.html). However, when I run `composer require aws/aws-sdk-php`, it gives:

Your requirements could not be resolved to an installable set of packages.
  Problem 1
    - pimcore/pimcore is locked to version v6.8.10 and an update of this package was not requested.
    - pimcore/pimcore v6.8.10 requires ext-dom * -> it is missing from your system. Install or enable PHP's dom ext
ension.
  Problem 2
    - codeception/codeception is locked to version 2.4.5 and an update of this package was not requested.
    - codeception/codeception 2.4.5 requires ext-mbstring * -> it is missing from your system. Install or enable PH
P's mbstring extension.
  Problem 3
    - aws/aws-sdk-php[3.173.0, ..., 3.173.22] require ext-simplexml * -> it is missing from your system. Install or
 enable PHP's simplexml extension.
    - Root composer.json requires aws/aws-sdk-php ^3.173 -> satisfiable by aws/aws-sdk-php[3.173.0, ..., 3.173.22].
To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.4/cli/php.ini
    - /etc/php/7.4/cli/conf.d/10-opcache.ini
    - /etc/php/7.4/cli/conf.d/10-pdo.ini
    - /etc/php/7.4/cli/conf.d/20-calendar.ini
    - /etc/php/7.4/cli/conf.d/20-ctype.ini
    - /etc/php/7.4/cli/conf.d/20-exif.ini
    - /etc/php/7.4/cli/conf.d/20-ffi.ini
    - /etc/php/7.4/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.4/cli/conf.d/20-ftp.ini
    - /etc/php/7.4/cli/conf.d/20-gettext.ini
    - /etc/php/7.4/cli/conf.d/20-iconv.ini
    - /etc/php/7.4/cli/conf.d/20-json.ini
    - /etc/php/7.4/cli/conf.d/20-phar.ini
    - /etc/php/7.4/cli/conf.d/20-posix.ini
    - /etc/php/7.4/cli/conf.d/20-readline.ini
    - /etc/php/7.4/cli/conf.d/20-shmop.ini
    - /etc/php/7.4/cli/conf.d/20-sockets.ini
    - /etc/php/7.4/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.4/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.4/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.4/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.

Does anyone know why this happened?

**Comment by aweichler** [comment, marked as answer]
Hi.
I think you need to install the PHP extensions simplexml, mbstring and dom.
In Ubuntu for example you can do this with apt install php7.4-xml php7.4-mbstring. Maybe the packeages does have an other name (php-mbrstring php-xml).

**Reply by alonr619** [reply]
Thanks, this worked for me.
